app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :favorite, :unfavorite]
+  before_action :block_unauthorized_changes, only: [:edit, :update, :destroy]
   before_filter :authenticate_user!
   # GET /posts
-  # GET /posts.json
   def index
-    @posts = Post.all.includes(:user)
+    # @posts = Post.all.includes(:user)
```

Comment:
Noted. I was experimenting and forgot to remove the line. I've removed it now. Thanks.